src/utils/hibp.js

```diff
@@ -147,8 +155,32 @@ async function loadBreachesIntoApp (app) {
   log.info('done-loading-breaches', 'great success 👍')
 }
 
+async function downloadBreachIcons (breaches) {
+  const breachDomains = breaches.map(breach => breach.Domain)
+  const logoFolder = resolve(dirname(fileURLToPath(import.meta.url)), '../client/images/logo_cache/')
+  try {
+    await mkdir(logoFolder)
```

Comment:
Are there any Kubernetes implications here, since we run multiple instances of the server?

Reply:
@mansaj? I don't know much about Kubernetes, though I'd expect the logo cache to just be recreated on every server?

Reply:
drive by 👀 , but I'm a bit confused as to when/how this gets called. In the current iteration, we merge some PRs and it gets pushed to a CDN.
But how frequently is the server polling for changes to the upstream HIBP feed and potentially fetch these logos via DDG?
Currently [HIBP](https://haveibeenpwned.com/api/v3/breaches) has 656 breaches. Is Monitor syncing w/ HIBP every 15 minutes? 60 minutes? The .env-dist seems to have a default of 10 minute polling, but not sure what you're using in prod.
https://github.com/mozilla/blurts-server/blob/56f08119dd546f880cefd18c4de8eebc47c34964/.env-dist#L48-L49

I didn't see if the code below was checking if the cached image is already stored locally, versus looping over `breachDomains.map(...)` and using `https.get()` and write streams to cache the .ico. So I guess wondering if we're fetching 650 .ico files every 10 minutes in production, that means roughly 3900 images per hour (per instance if we're caching locally and not on a shared CDN).
Or maybe add a quick check to see if we have a cached logo before refetching.

Reply:
Good callout, here's a check to only download it if we don't have an icon cached locally: ba354c12

Reply:
@Vinnl so I'm not sure if this directory would work when our service is running in kube. In our build steps we transform the `client` folder into `dist` folder. When this function gets called (on server startup), it will download the logos into the `client` folder, which isn't going to be used anymore.